design/ceph/object/bucketpolicy.md

```diff
@@ -0,0 +1,97 @@
+
+## Proposal details
+
+The following settings are needed to add for defining policies:
+  - _bucketPolicy_ in the `Spec` section of `CephObjectStoreUser` CR
+  - _bucketPolicy_ in the `parameters` section of `StorageClass` for `ObjectBucketClaim`
+
+Policies need to provide in generic `json` format. A policy can have multiple `statement`.
+
+The following checks are needed to perform in Rook to verify whether the `bucketpolicy` applicable to OBC.
+- for ceph object user, `Principal` value should have username and `Resource` should have specific bucket names. It can be defined for buckets which are not part of the OBC as well, the `bucketname` of an OBC can be fetched from its `configmap`.
+- In `StorageClass`, `Principal` value should be `*`(applicable to all users) and `Resource` should have the bucket name can be empty since it can be generated name from Rook as well and will be attached before setting the policy.
+
+Examples:
+
+```yaml
+# storageclass-bucket.yaml
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+   name: rook-ceph-delete-bucket
+provisioner: rook-ceph.ceph.rook.io/bucket
+reclaimPolicy: Delete
+parameters:
+  objectStoreName: my-store
+  objectStoreNamespace: rook-ceph
+  region: us-east-1
+  bucketName: ceph-bkt
+  bucketPolicy: "Version": "2012-10-17",
+                "Statement": [
+                {
+                "Sid": "listobjs",
+                "Effect": "Allow",
+                "Principal": {"AWS": ["arn:aws:iam:::*"]},
+                "Action": "s3:ListObject",
+                "Resource": "arn:aws:s3:::/*"
+                }
+                ]
+
+# object-user.yaml
+apiVersion: ceph.rook.io/v1
+kind: CephObjectStoreUser
+metadata:
+  name: my-user
+  namespace: rook-ceph
+spec:
+  store: my-store
+  displayName: "my display name"
+  bucketPolicy: "Version": "2012-10-17",
+                "Statement": [
+                {
+                "Sid": "putobjs"
+                "Effect": "Allow",
+                "Principal": {"AWS": ["arn:aws:iam:::my-user"]},
+                "Action": "s3:PutObject",
+                "Resource": "arn:aws:s3:::ceph-bkt-1/*"
+                },
+                {
+                "Sid": "getobjs"
+                "Effect": "Allow",
+                "Principal": {"AWS": ["arn:aws:iam:::my-user"]},
+                "Action": "s3:GettObject",
+                "Resource": "arn:aws:s3:::ceph-bkt-2/*"
+                }
+                ]
+```
+In the above examples, the `bucket policy` mentioned in the `storage class` will be inherited to all the OBCs created from it. And this policy need to be for the anonymous users(all users in the ceph object store), it will attached to bucket during the OBC creation.
```

Comment:
```suggestion
In the above examples, the `bucket policy` mentioned in the `storage class` will be inherited to all the OBCs created from it. And this policy needs to be for the anonymous users(all users in the ceph object store), it will be attached to the bucket during the OBC creation.
```

Reply:
Done